webapp/channels/src/components/admin_console/ip_filtering/index.tsx

```diff
@@ -0,0 +1,265 @@
+            saveConfirmModalProps.includeDisclaimer = false;
+        } else {
+            saveConfirmModalProps.title = formatMessage({id: 'admin.ip_filtering.apply_ip_filter_changes', defaultMessage: 'Apply IP Filter Changes'});
+            saveConfirmModalProps.subtitle = (
+                <FormattedMessage
+                    id={'admin.ip_filtering.apply_ip_filter_changes_are_you_sure'}
+                    defaultMessage={'Are you sure you want to apply these IP Filter changes? <strong>Users with IP addresses outside of the IP ranges provided will no longer have access to the workspace.</strong>'}
+                    values={{
+                        strong: (content: string) => <strong>{content}</strong>,
+                    }}
+                />
+            );
+            saveConfirmModalProps.buttonText = formatMessage({id: 'admin.ip_filtering.apply_changes', defaultMessage: 'Yes, apply changes'});
+            saveConfirmModalProps.includeDisclaimer = true;
+        }
+
+        setSaveConfirmationModal(<SaveConfirmationModal {...saveConfirmModalProps}/>);
+    }
+
+    const saveBarError = () => {
+        if (currentIPIsInRange) {
+            return undefined;
+        }
+
+        return (
+            <>
+                <AlertOutlineIcon size={16}/> {formatMessage({id: 'admin.ip_filtering.error_on_page', defaultMessage: 'Your IP address is not included in your filters'})}
+            </>
+        );
+    };
+
+    return (
+        <div className='IPFiltering wrapper--fixed'>
+            <AdminHeader>
+                {formatMessage({id: 'admin.ip_filtering.ip_filtering', defaultMessage: 'IP Filtering'})}
+            </AdminHeader>
+            <div className='MainPanel admin-console__wrapper'>
+                <>
+                    <EnableSectionContent
+                        filterToggle={filterToggle}
+                        setFilterToggle={setFilterToggle}
+                    />
+                    {ipFilters !== null && currentUsersIP !== null && filterToggle &&
+                        <EditSection
+                            ipFilters={ipFilters}
+                            currentUsersIP={currentUsersIP}
+                            setShowAddModal={setShowAddModal}
+                            setEditFilter={setEditFilter}
+                            handleConfirmDeleteFilter={handleConfirmDeleteFilter}
+                            currentIPIsInRange={currentIPIsInRange}
+                        />
+                    }
+                </>
+            </div>
+            {
+                editFilter !== null &&
+                <IPFilteringAddOrEditModal
+                    currentIP={currentUsersIP!}
+                    onClose={() => setEditFilter(null)}
+                    onSave={handleEditFilter}
+                    existingRange={editFilter!}
+                />
+            }
+            {
+                showAddModal &&
```

Comment:
Instead of rendering these modals directly, it might be easier to open them with the the `openModal` action like
```
dispatch(openModal({
    dialogType: IPFilteringAddOrEditModal,
    dialogProps: {
        currentIP: currentUsersIP,
        // onClose can be omitted and replaced with the onExited prop that the ModalController provides
        onSave: handleEditFilter,
        existingRange: editFilter,
    },
}));
```

Doing that means you won't need state to store the visibility of the modal or its props in this component's state